File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EErrorContainerLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EErrorContainerLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl43
            AppTheme.STANDARD -> R.color.cstl43
            AppTheme.BLUE -> R.color.cbl43
            AppTheme.GREEN -> R.color.cgl43
            AppTheme.RED -> R.color.crl43
            AppTheme.NOTHING -> R.color.cnol43
        }
    }
}

File: lib/layer2_core/light/e_error_container_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EErrorContainerLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl43;
      case 'standard': return cstl43;
      case 'blue': return cbl43;
      case 'green': return cgl43;
      case 'red': return crl43;
      case 'nothing': return cnol43;
      default: return cstl43;
    }
  }
}
